controllers/users.js

```diff
@@ -1,28 +1,21 @@
-const validator = require('validator');
 const jwt = require('jsonwebtoken');
 const bcrypt = require('bcryptjs');
 const User = require('../models/user');
 const NotFoundError = require('../errors/not-found-err');
-const ServerError = require('../errors/server-error');
 const AuthorError = require('../errors/AuthorizationError');
-const ValidationError = require('../errors/validation-error');
-require('dotenv').config();
 
 const { NODE_ENV, JWT_SECRET } = process.env;
 
 const login = (req, res, next) => {
   const { email, password } = req.body;
   User.findOne({ email })
     .select('+password')
-    .then((user) => {
-      if (!user) throw new AuthorError('Неправильные почта или пароль');
-      return Promise.all([
-        bcrypt.compare(password, user.password),
-        user,
-      ]);
-    })
+    .orFail(new AuthorError('Неправильные почта или пароль'))
+    .then((user) => Promise.all([
+      bcrypt.compare(password, user.password),
+      user,
+    ]))
     .then((array) => {
```

Comment:
Можно лучше: упростить выражение.
Уже писал об этом. `Promise.all()` здесь не нужен:
```suggestion
    .then((user) => bcrypt.compare(password, user.password))
    .then((result) => {
```

В `result` будет результат сравнения паролей. Значение `true/false`.

Reply:
исправлено